### PR TITLE
Fix generation of birthday, deathdate and anniversary

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Fix generation of birthday, deathdate and anniversary in case where no year is set - which is allowed as per https://tools.ietf.org/html/rfc6350#section-6.2.5

## How Has This Been Tested?
Unit tests are included

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: apps/dav/lib/CalDAV/BirthdayService.php [hunk 1]
```diff
@@ -119,26 +124,55 @@ public function buildDateFromContact($cardData, $dateField, $summarySymbol) {
...
+			$unknownYear = true;
+		}
+
 		try {
 			$date = new \DateTime($birthday);
 		} catch (Exception $e) {
 			return null;
 		}
+		if ($unknownYear) {
+			$summary = $doc->FN->getValue() . ' (' . $summarySymbol . '????)';
```
Comment (PVince81) [inline apps/dav/lib/CalDAV/BirthdayService.php]: is this summary displayed anywhere ? If yes we might want to just remove the "????" as it makes it look like some value is missing or that it's an error. However, from my understanding, omitting the year is an acceptable use case.

**Comment by PVince81** [comment]
Code looks good otherwise.

File: apps/dav/lib/CalDAV/BirthdayService.php (hunk 1 above)
Comment (PVince81) [inline apps/dav/lib/CalDAV/BirthdayService.php]: What's that symbol ?

Yeah, I'd say if the year is unknown, just don't add it to the summary.

**Comment by DeepDiver1975** [comment]
> 0270

no - the year 270 - when Nikolaus was born https://en.wikipedia.org/wiki/Saint_Nicholas - I need to get rid of the trailing 0 :see_no_evil:

**Comment by PVince81** [comment]
Ok, but if the year is unknown, what info do we have, only day and month ? Like second january would just be `(*01/02)` without year and `(*01/02/2017)` with year

**Comment by DeepDiver1975** [comment]
> Ok, but if the year is unknown, what info do we have, only day and month ? Like second january would just be (*01/02) without year and (*01/02/2017) with year

the event is displayed in the calendar on the birthday - no need to add day and month form my pov

**Comment by PVince81** [comment]
Aha, now I get it.

So either the summary displays the year `(*2017)` or if the year is not avaible, you wanted to display `(*????)` instead. I suggest to remove the parenthesis completely if the year is unknown, or display as `(*)` ?

**Comment by DeepDiver1975** [comment]
> Aha, now I get it.

sorry for not giving enough detailed information ...

**Comment by PVince81** [comment]
:+1:

**Pull request merged**